sushichef.py

```diff
@@ -1,33 +1,264 @@
+    """
+    Strucutre nodes from dictionary to connect them
+    """
+    for topic in dictionary:
+        if isinstance(topic, str):
+            topic_source_id = "{}-{}".format(prefix, topic.strip().replace(" ", "_"))
+            topic_node = nodes.TopicNode(source_id=topic_source_id, title=topic.capitalize())
+            LOGGER.info("\tCreating a math topic node - {}".format(topic))
+            get_nodes_from_dict(topic_node, dictionary[topic], topic_source_id)
+            parent.add_child(topic_node)
+        else:
+            for node in topic:
+                if node and not isinstance(node, str):
+                    LOGGER.info("\tAdding a child node - {}".format(node))
+                    parent.add_child(node)
+
 
-class BetterWorldEdChef(LineCook):
+def get_info(information):
     """
-    Sushi chef for uploading the Better World Ed content folders generated by `souschef.py`.
+    Retrieves title of the source
+    For example, from the information below,
+    HYPERLINK("https://drive.google.com/open?id=0B9q-Bz2y-5byRWZwRHptZmk0eU0","Baby And Her Health")
+    Returns `Baby And Her Health`
+    """
+    information = information.lower()
+    if information.startswith("=hyperlink"):
+        splits = information.split(",")
+        return splits[1].split("\"")[1].strip()
+    return information.strip()
+
+def download_video(link):
     """
-    pass  # no custom code needed, just using generic `LineCook`
+    downloads video from the link
+    and returns video node
+    """
+    if "=HYPERLINK" not in link:
+        return link
+    splits = link.split(",")
+    video_link = splits[0].split("=HYPERLINK(\"")[1][:-1]
+    video_id = video_link.split("/")[-1]
+    ydl = youtube_dl.YoutubeDL({
+        'outtmpl': './downloads/%(id)s.%(ext)s',
+        'writeautomaticsub': True
+    })
+
+    with ydl:
+        try:
+            result = ydl.extract_info(video_link)
+        except:
+            LOGGER.info("\tThere was an error while downloding video - {}".format(video_link))
+            return None
+
+    if 'entries' in result:
+        video = result['entries'][0]
+    else:
+        video = result
 
+    video_title = video["title"].capitalize()
+    video_source_id = video_title.strip().replace(" ", "_")
+    video_path = "{}/{}.mp4".format(DOWNLOAD_DIRECTORY, video_id)
+    video_file = files.VideoFile(path=video_path, language=languages.getlang('en').code)
+    unique_id = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(5))
+    LOGGER.info("\tCreating a video node - {}".format(video_title))
+    video_node = nodes.VideoNode(
+        source_id="{}-video-{}".format(video_source_id, unique_id),
+        title=video_title,
+        files=[video_file],
+        license=CHANNEL_LICENSE
+    )
+    return video_node
+
+def download_document(link):
+    """
+    Downloads pdf from the link
+    and returns document node
+    """
+    if "=HYPERLINK" not in link:
```

Comment:
Could you add comments on this? It seems like it could be a useful function to reuse in other chefs

Reply:
Sounds good. I updated this in 36652dd